libr/cons/grep.c

```diff
@@ -589,13 +589,18 @@ R_API int r_cons_grepbuf(char *buf, int len) {
 				if (show) {
 					if (cons->grep_highlight && grep->str) {
 						char *str = r_str_ndup (tline, ret);
-						char *newstr = r_str_newf (Color_INVERT"%s"Color_RESET, grep->str);
+						char *grepstr = r_str_ndup (tline, ret);
+						char *old_grepstr = grepstr;
+						grepstr = (char *) r_str_casestr (grepstr, grep->str);
+						grepstr[strlen (grep->str)] = '\0';
+						char *newstr = r_str_newf (Color_INVERT"%s"Color_RESET, grepstr);
 						if (str && newstr) {
-							str = r_str_replace (str, grep->str, newstr, 1);
```

Comment:
better implement r_str_replace_icase() and put all the logic inside this function